MonoGame.Framework/Graphics/Texture2D.DirectX.cs

```diff
@@ -205,26 +206,36 @@ private int CalculateSubresourceIndex(int arraySlice, int level)
             return arraySlice * _levelCount + level;
         }
 
-        private static Texture2D PlatformFromStream(GraphicsDevice graphicsDevice, Stream stream)
+        private unsafe static Texture2D PlatformFromStream(GraphicsDevice graphicsDevice, Stream stream)
         {
-            if (!stream.CanSeek)
-                throw new NotSupportedException("stream must support seek operations");
+            var reader = new ImageReader();
+            int width, height, channels;
 
-            // For reference this implementation was ultimately found through this post:
-            // http://stackoverflow.com/questions/9602102/loading-textures-with-sharpdx-in-metro 
-            Texture2D toReturn = null;
-            SharpDX.WIC.BitmapDecoder decoder;
+            // The data returned is always four channel BGRA
+            var data = reader.Read(stream, out width, out height, out channels, Imaging.STBI_rgb_alpha);
 
-            using (var bitmap = LoadBitmap(stream, out decoder))
-            using (decoder)
+            // XNA blacks out any pixels with an alpha of zero.
+            if (channels == 4)
```

Comment:
This should be unnecessary as we always read 4 channels?

Reply:
Oh wait... is this the original number for channels before conversion to RGBA?  If so i see this is just here to avoid doing the black out if the data was 24bit and didn't have alpha to begin with.